### PR TITLE
Update price string to allow promotion text in paper order summary

### PR DESCRIPTION
**Pull request by lucymonie**
## What are you doing in this PR?
Extending the paper order summary price string to accommodate promotions.

[**Trello Card**](https://trello.com/c/f6MEbgZQ/3706-print-campaign-print-checkout-order-summary-does-not-include-promotion-discount-copy)

## Why are you doing this?
To make it possible to accurately describe the price customers will pay during the June sale.

## Screenshots
Screen Shot 2021-06-03 at 17 39 14

Screen Shot 2021-06-03 at 17 21 15

### CODE REVIEW

File: support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx [hunk 1]
```diff
@@ -63,14 +63,24 @@ function PaperOrderSummary(props: PropTypes) {
...
-
+  // This allows us to get the price without promotion, so we can say what the price will revert to
+  const paperWithoutPromo = getProductPrice(
+    props.productPrices,
+    props.fulfilmentOption,
+    props.productOption,
+  );
+  const monthsDiscounted = paperWithoutPromo.promotions &&
+    paperWithoutPromo.promotions.length && paperWithoutPromo.promotions[0].numberOfDiscountedPeriods;
+  const priceHasPromo = paperWithoutPromo.promotions && paperWithoutPromo.promotions.length > 0;
```
Comment (johnduffell) [inline support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx:74]:
```suggestion
  const priceHasPromo = paperWithoutPromo.promotions && paperWithoutPromo.promotions.length > 0;
  const monthsDiscounted = priceHasPromo && paperWithoutPromo.promotions[0].numberOfDiscountedPeriods;
```

Comment (lucymonie) [reply support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx:74]: Nice

**Review by johnduffell — APPROVED** [review]
looks good to me! I think the defensive coding for the optional promotions could be slightly simpler but nothing massive. 👍

File: support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx (hunk 1 above)
Comment (lucymonie) [reply support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx:74]: But sadly it just doesn't work 🤷🏻

File: support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ import { paperProductsWithoutDigital, type ProductOptions } from 'helpers/produc
 import { Collection, type FulfilmentOptions } from 'helpers/productPrice/fulfilmentOptions';
 import type { ActivePaperProducts } from 'helpers/productPrice/productOptions';
 import type { WithDeliveryCheckoutState } from 'helpers/subscriptionsForms/subscriptionCheckoutReducer';
-import { getPriceWithDiscount } from 'helpers/productPrice/paperProductPrices';
+import { getPriceWithDiscount, getProductPrice } from 'helpers/productPrice/paperProductPrices';
```
Comment (i-hardy) [inline support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx:14]:
```suggestion
import { getPriceWithDiscount, getProductPrice } from 'helpers/productPrice/paperProductPrices';
import {
  getAppliedPromo,
} from 'helpers/productPrice/promotions';
```

File: support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx [hunk 3]
```diff
@@ -63,14 +63,24 @@ function PaperOrderSummary(props: PropTypes) {
...
+  const paperWithoutPromo = getProductPrice(
+    props.productPrices,
+    props.fulfilmentOption,
+    props.productOption,
+  );
+  const monthsDiscounted = paperWithoutPromo.promotions &&
+    paperWithoutPromo.promotions.length && paperWithoutPromo.promotions[0].numberOfDiscountedPeriods;
+  const priceHasPromo = paperWithoutPromo.promotions && paperWithoutPromo.promotions.length > 0;
+  const promotionPriceString = priceHasPromo && monthsDiscounted
+    ? ` for ${monthsDiscounted} months, then ${showPrice(paperWithoutPromo, false)} per month` : '';
```
Comment (i-hardy) [inline support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx:76]:
```suggestion
  const activePromo = getAppliedPromo(paperWithoutPromo.promotions);
  const monthsDiscounted = activePromo && activePromo.numberOfDiscountedPeriods;
  const promotionPriceString = activePromo && monthsDiscounted
    ? ` for ${monthsDiscounted} months, then ${showPrice(paperWithoutPromo, false)} per month` : '';
```

Just a couple of suggestions to tidy up this logic a bit- the getAppliedPromo function already discerns whether there's a relevant promotion that we can get information from.

File: support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx (hunk 1 above)
Comment (johnduffell) [reply support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx:74]: that's the magic of javascript... 🌩️

File: support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx (hunk 3 above)
Comment (i-hardy) [reply support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx:76]: Oh whoops I was too slow 😅

Comment (lucymonie) [reply support-frontend/assets/pages/paper-subscription-checkout/components/orderSummary/orderSummary.jsx:76]: Thanks for this - I'm going to do a tiny PR to fix this - coming your way shortly!

**Pull request merged**